File: json/main_processor.c
#include <stdio.h>
#include <time.h>

#include "sensors.h"

int main() {
  // Create a time_t variable to store the current time
  time_t raw_time;
  struct tm *time_info;
  char time_str[30];  // Buffer to store the formatted time string

  // Get the current time
  time(&raw_time);

  // Convert the time to local time (you can use gmtime() for UTC time)
  time_info = localtime(&raw_time);

  // Format the time as "YYYY-MM-DD HH:MM:SS"
  strftime(time_str, sizeof(time_str), "%Y-%m-%d %H:%M:%S", time_info);

  // Open a file to write the JSON data
  FILE *file = fopen("sensor_data.json", "w");
  if (file == NULL) {
    printf("Error opening file!\n");
    return 1;
  }

  // Simulating sensor data
  float temperature = get_temperature();
  float humidity = get_humidity();

  // Write the formatted data into the file
  fprintf(file,
          "{\n  \"time\": \"%s\",\n  \"temperature\": %.2f,\n  \"humidity\": "
          "%.2f\n}\n",
          time_str, temperature, humidity);

  // Close the file
  fclose(file);

  printf("Data successfully written to file.\n");

  return 0;
}

File: json/main_processor.py
import time
import json

# from get_uv import get_uv_data


# Simulating sensor data (replace these functions with actual sensor reading logic)
def get_temperature():
    # This is a placeholder for actual temperature reading logic
    return 24.5


def get_humidity():
    # This is a placeholder for actual humidity reading logic
    return 65.3


def main():
    # Get the current time in "YYYY-MM-DD HH:MM:SS" format
    current_time = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())

    # Gather sensor data
    temperature = get_temperature()
    humidity = get_humidity()
    # uv = get_uv_data()

    # Create the data dictionary to be written to the JSON file
    sensor_data = {
        "time": current_time,
        "temperature": temperature,
        "humidity": humidity,
    }

    # Write the JSON data to a file
    with open("sensor_data.json", "w") as file:
        json.dump(sensor_data, file, indent=2)

    print("Data successfully written to sensor_data.json")


if __name__ == "__main__":
    main()
